draft-ietf-ppm-dap.md

```diff
@@ -1305,50 +1493,89 @@ given batch interval, denoted `batch_interval`, decryption works as follows:
+First the aggregator checks that the request respects any batch "boundaries"
+determined by the query configuration:
+
+* For time-interval tasks, the query configuration specifies the "minimum batch
+  duration", which is determined by `time_precision`. For the `batch_interval`
+  included with the query, the Aggregator checks that both
+  `batch_interval.start` and `batch_interval.duration` are divisible by
+  `time_precision` and that `batch_interval.duration >= time_precision`.
+
+* For fixed-size tasks, the batch is assocaited with a "batch ID" selected by
+  the Leader. Thus the Aggregator needs to check that the query is associated
+  with a known batch ID:
+
+    * If the query contained in the CollectReq has type `by-batch-id`, the
+      Leader checks that the provided batch ID corresponds to a batch ID it
+      returned in a previous CollectResp for the task.
+
+    * The Helper checks that the batch ID provided by the leader in its
+      AggregateShareReq corresponds to a batch ID used in a previous
+      AggregateInitializeReq for the task.
+
+If the boundary check fails, then the Aggregator MUST abort with error
+"batchInvalid".
+
+Next, the Aggregator checks that batch contains a valid number of reports. Let
+`X` denote the set of reports in the batch:
+
+* For time-interval tasks, the query configuration specifies the minimum batch
+  size, `min_batch_size`. The Aggregator checks that `len(X) >= min_batch_size`.
+
+* For fixed-size tasks, the query configuration specifies the minimum batch
+  size, `min_batch_size`, and maximum batch size. The Aggregator checks that
+  `len(X) >= min_batch_size` and `len(X) <= max_batch_size`.
+
+If the batch size check fails, then the Aggregator MUST abort with error
+"invalidBatchSize".
+
+Next, the Aggregator checks that the batch has not been aggregated too many
+times. This is determined by the maximum batch lifetime, `max_batch_lifetime`.
+Unless the query has been issued less than `max_batch_lifetime` times, the
+Aggregator MUST abort with error "batchLifetimeExceeded".
+
+Finally, the Aggregator checks that the batch does not contain a report that was
+included in any previous batch. If this batch intersection check fails, then the
+Aggregator MUST abort with error "batchOverlap".
+
+Depending on the query type, properly checking for batch intersections may be
+prohibitively expensive for some deployments. The following is sufficient (but
+not necessary):
+
+* For time-interval tasks, the Aggregator checks that the batch interval does not
+  overlap with the batch interval of any previous query. If this batch interval
+  check fails, then the Aggregator MAY abort with error "batchOverlap".
 
 [[OPEN ISSUE: #195 tracks how we might relax this constraint to allow for more
 collect query flexibility. As of now, this is quite rigid and doesn't give the
 collector much room for mistakes.]]
 
+[OPEN ISSUE: We might need to permit the Aggregator to drop a request as needed, if
```

Comment:
Definitely. Why is this even an open issue?

Reply:
Yup, this appears to be redundant: see https://github.com/ietf-wg-ppm/draft-ietf-ppm-dap/issues/180. Thanks @tgeoghegan!